tests/rptest/tests/leaders_info_api_test.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2022 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from rptest.services.cluster import cluster
+from rptest.clients.types import TopicSpec
+from rptest.services.admin import Admin
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.services.rpk_producer import RpkProducer
+from rptest.tests.end_to_end import EndToEndTest
+
+from ducktape.utils.util import wait_until
+
+
+class LeadersInfoApiTest(EndToEndTest):
+    def __init__(self, test_context):
+        super(LeadersInfoApiTest, self).__init__(test_context=test_context)
+
+    @cluster(num_nodes=5)
+    def rest_leaders_info_test(self):
+        self.start_redpanda(num_nodes=3)
+        self.admin = Admin(self.redpanda)
+
+        self.spec = TopicSpec(partition_count=3, replication_factor=3)
+
+        self.client().create_topic(self.spec)
+        self.topic = self.spec.name
+
+        self.start_producer(1, throughput=100)
+        self.start_consumer(1)
+        self.await_startup(min_records=100, timeout_sec=180)
+
+        def check_reset_leaders():
+            for node in self.redpanda.nodes:
+                self.admin.reset_leaders_info(node)
+
+            partition_without_leader = 0
+            for partition in range(self.spec.partition_count):
+                leader = self.admin.get_partition_leader(namespace="kafka",
```

Comment:
By default, the admin client is going to pick a random node to query.  If you just wanted to check that the node that was reset, you could update `get_partition_leader` to take a node argument and do it explicitly.